L1Trigger/L1TTrackMatch/plugins/L1TrackJetClustering.h

```diff
@@ -0,0 +1,367 @@
+
+      my_pt = phislice[etabin].pTtot;
+      if (my_pt == 0)
+        continue;
+
+      //get previous bin pT
+      if (etabin > 0 && !phislice[etabin - 1].used)
+        previousbin_pt = phislice[etabin - 1].pTtot;
+
+      // get next bins pt
+      if (etabin < etaBins_ - 1 && !phislice[etabin + 1].used) {
+        nextbin_pt = phislice[etabin + 1].pTtot;
+        if (etabin < etaBins_ - 2 && !phislice[etabin + 2].used) {
+          nextbin2_pt = phislice[etabin + 2].pTtot;
+        }
+      }
+      // check if pT of current cluster is higher than neighbors
+      if (my_pt < previousbin_pt || my_pt <= nextbin_pt) {
+        // if unused pT in the left neighbor, spit it out as a cluster
+        if (previousbin_pt > 0) {
+          clusters.push_back(phislice[etabin - 1]);
+          phislice[etabin - 1].used = true;
+          nclust++;
+        }
+        continue;  //if it is not the local max pT skip
+      }
+      // here reach only unused local max clusters
+      clusters.push_back(phislice[etabin]);
+      phislice[etabin].used = true;  //if current bin a cluster
+      if (previousbin_pt > 0) {
+        clusters[nclust].pTtot += previousbin_pt;
+        clusters[nclust].ntracks += phislice[etabin - 1].ntracks;
+        clusters[nclust].nxtracks += phislice[etabin - 1].nxtracks;
+        for (unsigned int itrk = 0; itrk < phislice[etabin - 1].trackidx.size(); itrk++)
+          clusters[nclust].trackidx.push_back(phislice[etabin - 1].trackidx[itrk]);
+      }
+
+      if (my_pt >= nextbin2_pt && nextbin_pt > 0) {
+        clusters[nclust].pTtot += nextbin_pt;
+        clusters[nclust].ntracks += phislice[etabin + 1].ntracks;
+        clusters[nclust].nxtracks += phislice[etabin + 1].nxtracks;
+        for (unsigned int itrk = 0; itrk < phislice[etabin + 1].trackidx.size(); itrk++)
+          clusters[nclust].trackidx.push_back(phislice[etabin + 1].trackidx[itrk]);
+        phislice[etabin + 1].used = true;
+      }
+
+      nclust++;
+
+    }  // for each etabin
+
+    // Merge close-by clusters
+    for (int m = 0; m < nclust - 1; ++m) {
+      if (((clusters[m + 1].eta - clusters[m].eta) < (3 * etaStep_) / 2) &&
+          (-(3 * etaStep_) / 2 < (clusters[m + 1].eta - clusters[m].eta))) {
+        if (clusters[m + 1].pTtot > clusters[m].pTtot) {
+          clusters[m].eta = clusters[m + 1].eta;
+        }
+        clusters[m].pTtot += clusters[m + 1].pTtot;
+        clusters[m].ntracks += clusters[m + 1].ntracks;    // total ntrk
+        clusters[m].nxtracks += clusters[m + 1].nxtracks;  // total ndisp
+        for (unsigned int itrk = 0; itrk < clusters[m + 1].trackidx.size(); itrk++)
+          clusters[m].trackidx.push_back(clusters[m + 1].trackidx[itrk]);
+
+        // if remove the merged cluster - all the others must be closer to 0
+        for (int m1 = m + 1; m1 < nclust - 1; ++m1)
+          clusters[m1] = clusters[m1 + 1];
+
+        clusters.erase(clusters.begin() + nclust);
+        nclust--;
+        m = -1;
```

Comment:
This was removed in L1Clustering.h with #40387, and it looks like it has to be removed also here: please check